nionswift_plugin/usim/ScanDevice.py

```diff
@@ -150,7 +150,9 @@ def read_partial(self, frame_number, pixels_to_skip) -> (typing.Sequence[dict],
                     current_frame.bad = True
                     current_frame.complete = True
                 self.__instrument.live_probe_position = Geometry.FloatPoint(y=y / h, x=x / w)
-                target_count = current_frame.data_count + 1
+                sequence_progress = self.__instrument.sequence_progress
+                target_count = max(sequence_progress - 2*(sequence_progress//(size.width+2) + 1) + 1,
```

Comment:
I'm not certain that we should use 'max' here. I'm not sure it's worth holding up the commit, but it seems like we're either scanning with syncing to camera or not; and we could indicate the two modes by have another flag or using sequence_progress=None to mean "not synced with camera". Any thoughts?

Reply:
I think it might be good to have "max" there because in the ideal case, sequence_progress and target_count will be the same value. If the scan is too slow, the value of sequence_progress will be higher which is also the value we want in this case. I left target_count in there to make sure that we are not caught in an endless loop if something goes wrong with "sequence_count". The camera might be already stopped and not update it anymore. Then "wait_for_camera_frame" above will time out and the scan will end eventually because target_count is still increased.

Reply:
Ok great.